src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp

```diff
@@ -672,10 +673,12 @@ inline void multiply_scalar(float* a, ov::bfloat16* a_dst, const float val, cons
         __mmask16 mask = (1 << (size - i)) - 1;
         v_a = _mm512_maskz_loadu_ps(mask, a + i);
         v_a = _mm512_mul_ps(v_a, v_scale);
-        mm512_uni_mask_storeu_ps(a_dst + i, mask, v_a);
+        mm512_uni_storeu_tail_ps(a_dst + i, v_a, size - i);
+
+        i += (size - i);
     }
 #else
-    for (size_t i = 0; i < size; i++) {
+    for (; i < size; i++) {
```

Comment:
Why this change was done? Seems like vector code should handle all the elements.

Reply:
> Why this change was done? Seems like vector code should handle all the elements.

Actually, no any big change here. Just use a template to unify the `bfloat16` and `float16` code together.